src/share/classes/com/sun/javatest/regtest/agent/TestThreadFactory.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.sun.javatest.regtest.agent;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * {@code TestThreadFactory} allows some customization of test execution.
+ * The jtreg creates new thread for each test using {@code new Thread(ThreadGroup tg, Runnable task);}.
```

Comment:
Recommend: ~The~ `jtreg creates a new thread`

Reply:
The name is substantially better than before (it used to be `CustomMainWrapper`).

Now, the name is annoyingly ambiguous.   It can be read as `Test` (imperative verb) `Thread-Factory` (noun) or `Test-Thread` (adjective) `Factory` (noun). (I believe it is the latter that is intended.)

I can't think of a better name, and I don't want to suggest we should go through another name change again!
For now, I think it may be enough to add a new explanatory sentence at the beginning of the doc comment,
and/or work on the rest of the comment text.
```
A factory for threads used to run tests.
```